docs/client/client-overview.md

```diff
@@ -0,0 +1,177 @@
+
+## Project Configuration
+
+GraphQL Kotlin provides both Gradle and Maven plugins to automatically generate your client code at build time. Once
+your data classes are generated, you can then execute their underlying GraphQL operations using `graphql-kotlin-client`
+runtime dependency.
+
+Basic `build.gradle.kts` Gradle configuration:
+
+```kotlin
+import com.expediagroup.graphql.plugin.gradle.graphql
+
+plugins {
+    id("com.expediagroup.graphql") version $latestGraphQLKotlinVersion
+}
+
+dependencies {
+  implementation("com.expediagroup:graphql-kotlin-client:$latestGraphQLKotlinVersion")
+}
+
+graphql {
+    endpoint = "http://localhost:8080/graphql"
+    packageName = "com.example.generated"
+}
+```
+
+Equivalent `pom.xml` Maven configuration
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
+    <artifactId>graphql-kotlin-maven-client-example</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <properties>
+        <graphql-kotlin.version>$latestGraphQLKotlinVersion</graphql-kotlin.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.expediagroup</groupId>
+            <artifactId>graphql-kotlin-client</artifactId>
+            <version>${graphql-kotlin.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.expediagroup</groupId>
+                <artifactId>graphql-kotlin-maven-plugin</artifactId>
+                <version>${graphql-kotlin.version}</version>
+                <executions>
+                    <execution>
+                        <id>introspect-schema</id>
+                        <goals>
+                            <goal>introspectSchema</goal>
+                        </goals>
+                        <configuration>
+                            <endpoint>http://localhost:8080/graphql</endpoint>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>generate-client</id>
+                        <goals>
+                            <goal>generateClient</goal>
+                        </goals>
+                        <configuration>
+                            <packageName>com.example.generated</packageName>
+                            <schemaFile>${project.build.directory}/schema.graphql</schemaFile>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+```
+
+See [graphql-kotlin-client-example](https://github.com/dariuszkuc/graphql-kotlin-client-example) project for complete
```

Comment:
I could move this project under `/examples` but they won't be run during the build, i.e. in order to build clients they need a running server.